### PR TITLE
Obsolete ResponseMode in MicrosoftChallengeProperties

### PR DESCRIPTION
**Pull request by Kahbazi**
Deprecate `ResponseMode` since only the default value is supported in Microsoft Account authentication.

@Tratcher

### CODE REVIEW

File: src/Security/Authentication/MicrosoftAccount/src/MicrosoftChallengeProperties.cs [hunk 1]
```diff
@@ -11,6 +12,7 @@ public class MicrosoftChallengeProperties : OAuthChallengeProperties
         /// <summary>
         /// The parameter key for the "response_mode" argument being used for a challenge request.
         /// </summary>
+        [Obsolete("This parameter is not supported in MicrosoftAccountHandler.")]
```
Comment (Tratcher) [inline src/Security/Authentication/MicrosoftAccount/src/MicrosoftChallengeProperties.cs:15]: This caused some secondary build errors that need to be suppressed.

**Review by Tratcher — APPROVED** [review]

**Comment by Tratcher** [comment]
Thanks

**Comment by davidfowl** [comment]
We usually point to a url explaining why we deprecated something and provide an alternative.

**Comment by Tratcher** [comment]
In this case I don't think much explanation is warranted. The property was introduced by mistake, never worked for anything but the default value, can't be made to work, and was never reported as broken by a customer so it seems it was never used. The only alternative API is to switch to the OIDC package.

**Comment by davidfowl** [comment]
So in theory nobody ever used this property

**Comment by Tratcher** [comment]
Not that we know of. We only noticed the issue when updating the doc comments.

**Pull request merged**